internal/runtime/hcsv2/nvidia_utils.go

```diff
@@ -0,0 +1,87 @@
+// +build linux
+
+package hcsv2
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/Microsoft/opengcs/internal/storage/pci"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+// path that the shim mounts the nvidia gpu vhd to in the uvm
+// this MUST match the path mapped to in the shim
+const lcowNvidiaMountPath = "/run/nvidia"
+
+// annotation to find the gpu capabilities on the container spec
+// must match the hcsshim annotation string for gpu capabilities
+const annotationContainerGPUCapabilities = "io.microsoft.container.gpu.capabilities"
+const nvidiaDebugFilePath = "/nvidia-container.log"
+
+// TODO katiewasnothere: prestart hooks will be depracated, this needs to be moved to a createRuntime hook
+// described here: https://github.com/opencontainers/runtime-spec/blob/39c287c415bf86fb5b7506528d471db5405f8ca8/config.md#posix-platform-hooks
+// addNvidiaDevicePreHook builds the arguments for nvidia-container-cli and creates the prestart hook
+func addNvidiaDevicePreHook(ctx context.Context, spec *oci.Spec) error {
+	nvidiaToolBinary := "nvidiaPrestartHook"
+	nvidiaToolPath, err := exec.LookPath(nvidiaToolBinary)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find %s for container GPU support", nvidiaToolBinary)
+	}
+
+	debugOption := fmt.Sprintf("--debug=%s", nvidiaDebugFilePath)
+	args := []string{nvidiaToolPath, debugOption, "--load-kmods", "--no-pivot", "configure", "--ldconfig=@/sbin/ldconfig"}

```

Comment:
I remember the situation with ldconfig and whether we use the version on the host or in the container to be somewhat complicated. Can you add a comment explaining the reasoning behind what you're doing with that?

Reply:
Both the host and container ldconfig don't work as intended for the nvidia-container-cli tool here. So really until I can investigate that issue more, this is just arbitrary. 

Since this is still an open issue, the workaround is just to run ldconfig as part of the container config's command.

Reply:
If this isn't resolved prior to checking in, please make sure you add a comment explaining this.